src/components/selfCheckQuestion/index.tsx

```diff
@@ -0,0 +1,450 @@
+        React.useState(alphanumeric);
+    function changeAlphanumeric(e) {
+        setAlphanumericInput(e);
+        onChangeAlphanumeric(questionId, e);
+    }
+    const [selectedOption, setSelectedOption] = React.useState(type);
+    function questionType(e) {
+        setSelectedOption(e.target.value);
+        onChangeQuestionType(questionId, e.target.value);
+    }
+    const [sliderStart, setSliderStart] = React.useState(1);
+    const [sliderEnd, setSliderEnd] = React.useState(options.length);
+
+    function sliderLowerBound(e) {
+        setSliderStart(e.target.value);
+        onChangeSliderOption(questionId, e.target.value, sliderEnd);
+    }
+    function sliderUpperBound(e) {
+        setSliderEnd(e.target.value);
+        onChangeSliderOption(questionId, sliderStart, e.target.value);
+    }
+
+    return (
+        <Box overflowX="auto">
+            <Box
+                borderWidth="2px"
+                borderRadius="lg"
+                rounded={"md"}
+                p={6}
+                borderColor="#C0BABA"
+            >
+                <Flex wrap={"wrap"}>
+                    <Flex minWidth={"40%"} p="2">
+                        <Heading
+                            fontSize={20}
+                            fontWeight="500"
+                            alignSelf="center"
+                            mr={6}
+                        >
+                            {questionIndex + 1}.
+                        </Heading>
+                        <Input
+                            width={"full"}
+                            size={"lg"}
+                            variant="flushed"
+                            placeholder="Untitled Question"
+                            mr={6}
+                            pl={2}
+                            isTruncated
+                        />
+                    </Flex>
+                    <Flex pb={15} marginLeft={"auto"}>
+                        <Select
+                            value={selectedOption}
+                            minWidth={"165"}
+                            mr={selfCheckQuestionSize == 1 ? 0 : 6}
+                            onChange={(e) => questionType(e)}
+                        >
+                            {optionList.map((choice, index) => (
+                                <option
+                                    index={index}
+                                    key={choice.value + index + questionId}
+                                    value={choice.value}
+                                >
+                                    {choice.label}
+                                </option>
+                            ))}
+                        </Select>
+
+                        {questionIndex != 0 && (
+                            <IconButton
+                                onClick={() => onMoveUpQuestion(questionIndex)}
+                                ml={
+                                    questionIndex + 1 == selfCheckQuestionSize
```

Comment:
whats the purpose of this?

Reply:
I'm assuming this is to figure out whether there's a 'move question down' button needed or not

Reply:
On the very last question, there should only be the move up button. As a result the move down button is not shown, as a result the left margin must be increased to make sure everything is still lined up with the other questions